packages/remix-deferred/index.ts

```diff
@@ -0,0 +1,449 @@
+export const CONTENT_TYPE = "text/remix-deferred";
```

Comment:
I would love if `DeferredData` could be implemented in a more "webby" way, instead of relying on a custom Content-Type. Also, I totally get why you'd restrict `deferred` to JSON responses only as a first pass as this solves most use cases but it would be great if you could defer other content types eg. blobs and forms.

I think both of these needs can be met by changing `DeferredData`'s internal representation to use `Content-Type: multipart/mixed`, or maybe it should be `multipart/parallel` as deferred parts indeed do not have any particular order - they're just streamed when ready.

Because each part of a multipart response has it's own headers it allows `DeferredData` to contain many different content types, including nested `DeferredData`s. I'm not sure how the latter could be represented in JS (probably not possible with promises) but it's fun to think about.

The other benefit here is that you're not then inventing your own internal content type and you can take advantage of prior art such as https://github.com/maraisr/meros and https://github.com/tom-sherman/response-multipart. This has a benefit to consumers in that devtools can display multipart responses in a nice way which can aid debugging.

Would love to hear thoughts!

Reply:
Using `multipart/mixed` actually makes the response body super readable in plaintext:

```
... other headers here ...
Content-type: multipart/mixed; boundary="remix boundary" 
--remix boundary
Content-Type: application/json
Remix-Event-Key: data
Remix-Settled-Key: someData

{"count": 1}
--remix boundary
Content-Type: application/octet-stream
Remix-Event-Key: data
Remix-Settled-Key: anImage

... imagine some image data here ...

--remix boundary-- 
```

Note here I've used the custom HTTP headers `Remix-Settled-Key` and `Remix-Event-Key`, these are just examples of how a multipart body could carry the key name.

Reply:
Finally, just as a note, part of this request is because I feel like there is a lot of overlap with what you're doing with DeferredData and what others are doing with multipart bodies. Your `readStreamSections` implementation for example could be lifted and shifted into [my multipart library](https://github.com/tom-sherman/response-multipart) to support full streaming of each parts.

At the end of the day this is why standards and Using The Platform™️ are so great!